torchgeo/datasets/geo.py

```diff
@@ -418,6 +420,36 @@ def __init__(
         self._crs = cast(CRS, crs)
         self._res = cast(float, res)
 
+    def list_files(self, filename_glob: Optional[str] = None) -> list[str]:
```

Comment:
@calebrob6 Do you have any feedback? This file contains is the main contributions. The rest is mainly refactoring.